core/Plugin/Controller.php

```diff
@@ -477,7 +477,7 @@ protected function getLastUnitGraphAcrossPlugins($currentModuleName, $currentCon
     protected function getGraphParamsModified($paramsToSet = array())
     {
         if (!isset($paramsToSet['period'])) {
-            $period = Common::getRequestVar('period');
```

Comment:
Haven't had a closer look at the usage here. But we need to check everywhere we change that, that it might change the behavior. If no `vartype` is passed to `Common::getRequestVar` it simply returns `null` if the parameter can't be found in the request. The new methods are both passing `string` as `varType`, causing an exception if no default is provided.

Reply:
@sgiehl that make sense. Would that works 
```php
  try {
     try {
            $period = isset($paramsToSet['period']) ?? Piwik::getPeriod();
        } catch (\Exception $e) {
            $period = null;
        }
```